app/services/commitments/minimum/in_arrears/fetch_invoices_service.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Commitments
+  module Minimum
+    module InArrears
+      class FetchInvoicesService < Commitments::FetchInvoicesService
+        private
+
+        def dates_service
+          ds = Subscriptions::DatesService.new_instance(
+            subscription,
+            invoice_subscription.timestamp,
+            current_usage: subscription.terminated?,
+          )
+
+          return ds unless subscription.terminated?
+
+          Subscriptions::TerminatedDatesService.new(
+            subscription:,
+            invoice: invoice_subscription.invoice,
+            date_service: ds,
+          ).call
+        end
+
+        def fetch_invoices
+          plan = subscription.plan
+
+          if !subscription.plan.yearly? || !plan.bill_charges_monthly?
+            return Invoice.where(id: invoice_subscription.invoice_id)
+          end
+
+          invoice_ids = subscription
+            .invoice_subscriptions
+            .where(
+              'from_datetime >= ? AND to_datetime <= ?',
+              dates_service.previous_beginning_of_period,
+              dates_service.end_of_period,
+            ).pluck(:invoice_id)
+
+          Invoice.where(id: invoice_ids)
```

Comment:
You could have the same result with a single DB query
```suggestion
          invoice_ids_query = subscription
            .invoice_subscriptions
            .where(
              'from_datetime >= ? AND to_datetime <= ?',
              dates_service.previous_beginning_of_period,
              dates_service.end_of_period,
            ).select(:invoice_id)

          Invoice.where(id: invoice_ids_query)
```